JDBC/OAuthSampleApp/src/main/java/OAuthSampleApp.java

```diff
@@ -10,91 +10,309 @@
-    private String accessToken = "";
-
-    @Option(names = { "-r", "--refresh-token" }, description = "Refresh token")
-    private String refreshToken = "";
-
-    @Option(names = { "-s", "--client-secret" }, description = "Client Secret")
-    private String clientSecret = "";
-
-    private static Connection connectToDB(String host, String port, String dbName, String accessToken,
-            String refreshToken, String clientSecret) throws SQLException {
-        Properties jdbcOptions = new Properties();
-        jdbcOptions.put("oauthaccesstoken", accessToken);
-        jdbcOptions.put("oauthrefreshtoken", refreshToken);
-        jdbcOptions.put("oauthclientsecret", clientSecret);
-
-        return DriverManager.getConnection(
-                "jdbc:vertica://" + host + ":" + port + "/" + dbName, jdbcOptions);
-    }
-
-    private static ResultSet executeQuery(Connection conn) throws SQLException {
-        Statement stmt = conn.createStatement();
-        return stmt.executeQuery("SELECT user_id, user_name FROM USERS ORDER BY user_id");
-    }
-
-    private static void printResults(ResultSet rs) throws SQLException {
-        int rowIdx = 1;
-        while (rs.next()) {
-            System.out.println(rowIdx + ". " + rs.getString(1).trim() + " " + rs.getString(2).trim());
-            rowIdx++;
-        }
-    }
-
-    @Override
-    public Integer call() throws Exception {
-        try {
-            Connection conn = connectToDB(host, port, dbName, accessToken, refreshToken, clientSecret);
-            ResultSet rs = executeQuery(conn);
-            printResults(rs);
-            conn.close();
-        } catch (SQLTransientConnectionException connException) {
-            System.out.print("Network connection issue: ");
-            System.out.print(connException.getMessage());
-            System.out.println(" Try again later!");
-        } catch (SQLInvalidAuthorizationSpecException authException) {
-            System.out.print("Could not log into database: ");
-            System.out.print(authException.getMessage());
-            System.out.println(" Check the login credentials and try again.");
-        } catch (SQLException e) {
-            e.printStackTrace();
-        }
-        return 0;
-    }
-
-    public static void main(String[] args) {
-        int exitCode = new CommandLine(new OAuthSampleApp()).execute(args);
-        System.exit(exitCode);
-    }
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.FileInputStream;
+import java.io.BufferedInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.UnsupportedEncodingException;
+import java.nio.charset.StandardCharsets;
+import java.net.URLEncoder;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import com.google.gson.JsonParser;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.vertica.jdbc.VerticaConnection;
+
+public class OAuthSampleApp {
```

Comment:
Unless it is just rendering poorly in github, the spacing in this class is all off. Indentation should simply be four spaces per indentation level and they should be actual spaces, not tabs.

Reply:
Hi Danny, 
This indentation formatting is done using the auto-indent feature(Source->Format) in Eclipse. So If we make any changes here in git for indentation, It will reflect wrong when we open the file in Eclipse or other IDE.

Reply:
If you are going to use the auto indent feature you need to update it to use four spaces instead of tabs, there should be an option for this. A space is a space, it's the same everywhere. A tab character may not render the same everywhere like we see here

Reply:
Ok. As per last commit i think, All indentation is good and no issues observed. For next updates/commits, will change it to 4 spaces instead of tab.